### PR TITLE
Add functionality to export build timings.

### PR DESCRIPTION
**Pull request by Kelimion**
Usage:
`odin build examples/demo -show-extra-timings -export-timings:json -export-timings-file:timings.json`
`odin build examples/demo -show-timings -export-timings:csv -export-timings-file:timings.csv`

Example output:
```json
{
        "totals": [
                {"name": "total_packages",  "count": 23},
                {"name": "total_files",     "count": 114},
                {"name": "total_lines",     "count": 44413},
                {"name": "total_tokens",    "count": 257884},
                {"name": "total_file_size", "count": 1134058},
        ],
        "timings": [
                {"name": "Total Time", "millis": 1720.383},
                {"name": "initialization", "millis": 2.019},
                {"name": "parse files", "millis": 57.618},
                {"name": "type check", "millis": 316.631},
                {"name": "LLVM API Code Gen", "millis": 1239.287},
                {"name": "msvc-link", "millis": 104.822},
        ],
}
```

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -929,10 +933,10 @@ bool parse_build_flags(Array<String> args) {
 						}
 
 						if (ok) switch (bf.kind) {
-						case BuildFlag_Help:
+						case BuildFlag_Help: {
```
Comment (gingerBill) [inline src/main.cpp:936]: Please don't add blocks around cases when they were not needed in the first place.

File: src/main.cpp [hunk 2]
```diff
@@ -1508,6 +1567,107 @@ bool parse_build_flags(Array<String> args) {
...
+
+		gb_fprintf(&f, "\t],\n");
+
+		gb_fprintf(&f, "}\n");
+	}
+
+	/*
+		CSV export
+	*/
+	else if (build_context.export_timings_format == TimingsExportCSV) {
```
Comment (gingerBill) [inline src/main.cpp:1651]: Minor style issue: keep to the `} else if` style if possible and add the comment either after the `) {` or on the next line

File: src/timings.cpp [hunk 3]
```diff
@@ -169,14 +169,19 @@ f64 time_stamp(TimeStamp const &ts, u64 freq, TimingUnit unit) {
 	}
 }
 
-void timings_print_all(Timings *t, TimingUnit unit = TimingUnit_Millisecond) {
+void timings_print_all(Timings *t, TimingUnit unit = TimingUnit_Millisecond, bool timings_are_finalized = false) {
```
Comment (gingerBill) [inline src/timings.cpp:172]: Is there anything that sets `timings_are_finalized` to `true` at any point?

**Review by gingerBill — COMMENTED** [review]
Overall logic seems absolutely fine, just some minor style issues.

File: src/timings.cpp (hunk 3 above)
Comment (Kelimion) [reply src/timings.cpp:172]: Yes, `main.cpp:show_timings():1693` originally called this procedure and then finalized the timings. As I'm iterating over the timings again for the export, I don't want to stop the clock twice. So now it's called first for the stdout output as before, only with the implicit `false`, and then the export calling it sets it to `true`.

File: src/main.cpp (hunk 1 above)
Comment (Kelimion) [reply src/main.cpp]: Addressed in 8c5c45a04c2a525e048538d764cc35f44239230b.

File: src/timings.cpp (hunk 3 above)
Comment (gingerBill) [reply src/timings.cpp:172]: Okay. Makes sense!

**Pull request merged**